servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/ReplayStrategies.java

```diff
@@ -0,0 +1,382 @@
+     *     scheduling timer tasks.</li>
+     *     <li>{@code false} will evict expired items eagerly when they expire. If {@code ttl} is lower that
+     *     {@code historyHint} relative to signal arrival rate this can use less memory but schedules time tasks.
+     *     <p>
+     *     Note that timer expiration is done concurrently which may cause gaps in signal delivery to new subscribers.
+     *     For example:
+     *       <pre>
+     *         initial accumulation state: [onNext1, onNext2, onNext3, onNext4]
+     *         Thread1: add new subscriber, subscriber does request(4), deliver onNext1
+     *         Thread2: expire onNext1, onNext2, onNext3
+     *         Thread1: [deliver onNext2,] deliver onNext4
+     *           Note that onNext2 may or may not be delivered, and onNext3 was not delivered.
+     *       </pre>
+     *     </li>
+     * </ul>
+     * @param <T> The type of {@link ReplayStrategyBuilder}.
+     * @return a {@link ReplayStrategyBuilder} using the historyHint and TTL strategy.
+     */
+    public static <T> ReplayStrategyBuilder<T> historyTtlBuilder(int historyHint, Duration ttl, Executor executor,
+                                                                 boolean lazyEviction) {
+        return new ReplayStrategyBuilder<>(lazyEviction ?
+                () -> new LazyTimeLimitedReplayAccumulator<>(historyHint, ttl, executor) :
+                () -> new EagerTimeLimitedReplayAccumulator<>(historyHint, ttl, executor));
+    }
+
+    private static final class MostRecentReplayAccumulator<T> implements ReplayAccumulator<T> {
+        private final int maxItems;
+        private final Deque<Object> queue = new ArrayDeque<>();
+
+        MostRecentReplayAccumulator(final int maxItems) {
+            if (maxItems <= 0) {
+                throw new IllegalArgumentException("maxItems: " + maxItems + "(expected >0)");
+            }
+            this.maxItems = maxItems;
+        }
+
+        @Override
+        public void accumulate(@Nullable final T t) {
+            if (queue.size() >= maxItems) {
+                queue.poll();
+            }
+            queue.add(wrapNull(t));
+        }
+
+        @Override
+        public void deliverAccumulation(final Consumer<T> consumer) {
+            for (Object item : queue) {
+                consumer.accept(unwrapNullUnchecked(item));
+            }
+        }
+    }
+
+    private static final class LazyTimeLimitedReplayAccumulator<T> implements ReplayAccumulator<T> {
+        @SuppressWarnings("rawtypes")
+        private static final AtomicIntegerFieldUpdater<LazyTimeLimitedReplayAccumulator> queueSizeUpdater =
+                AtomicIntegerFieldUpdater.newUpdater(LazyTimeLimitedReplayAccumulator.class, "queueSize");
+        private final Executor executor;
+        private final Queue<TimeStampSignal<T>> items;
+        private final long ttlNanos;
+        private final int maxItems;
+        private volatile int queueSize;
```

Comment:
This is mostly the same as `MostRecentReplayAccumulator` now and I don't think we need to worry about concurrent calls to either `accumulate(..)` or `deliverAccumulation(..)` so we can use a standard queue and it's simple `.size()` method.